File: crm-backend/src/main/java/com/arjun/crm/service/AnalyticsService.java
package com.arjun.crm.service;

import com.arjun.crm.dto.response.ActivityAnalyticsResponse;
import com.arjun.crm.dto.response.TaskAnalyticsResponse;
import com.arjun.crm.dto.response.TeamPerformanceResponse;

import java.time.LocalDate;

public interface AnalyticsService {
    
    /**
     * Get task analytics for a specific workspace
     */
    TaskAnalyticsResponse getTaskAnalytics(Long workspaceId, LocalDate startDate, LocalDate endDate);
    
    /**
     * Get team performance analytics for a specific workspace
     */
    TeamPerformanceResponse getTeamPerformance(Long workspaceId, LocalDate startDate, LocalDate endDate);
    
    /**
     * Get activity analytics for a specific workspace
     */
    ActivityAnalyticsResponse getActivityAnalytics(Long workspaceId, LocalDate startDate, LocalDate endDate);
    
    /**
     * Get recent activities for a specific workspace
     */
    Object getRecentActivities(Long workspaceId, int limit);
}

File: crm-frontend/src/services/analyticsService.js
import api from './api'

const unwrap = (response) => response?.data ?? response

export const analyticsService = {
  getDashboard: async (workspaceId) => {
    const response = await api.get('/dashboard/overview', {
      params: { workspaceId }
    })
    return unwrap(response)
  },

  getTaskAnalytics: async (params) => {
    const response = await api.get('/analytics/tasks', { 
      params: { workspaceId: params.workspaceId, startDate: params.startDate, endDate: params.endDate } 
    })
    return unwrap(response)
  },

  getTeamPerformance: async (params) => {
    const response = await api.get('/analytics/team', { 
      params: { workspaceId: params.workspaceId, startDate: params.startDate, endDate: params.endDate } 
    })
    return unwrap(response)
  },

  getActivityAnalytics: async (params) => {
    const response = await api.get('/analytics/activity', { 
      params: { workspaceId: params.workspaceId, startDate: params.startDate, endDate: params.endDate } 
    })
    return unwrap(response)
  },

  getRecentActivities: async (workspaceId, limit = 10) => {
    const response = await api.get('/analytics/recent', { 
      params: { workspaceId, limit },
      silent: true 
    })
    return unwrap(response)
  },

  getDailyReport: async (date) => {
    const response = await api.get('/reports/daily', { params: { date } })
    return unwrap(response)
  },

  getWeeklyReport: async (startDate) => {
    const response = await api.get('/reports/weekly', { params: { startDate } })
    return unwrap(response)
  },

  getMonthlyReport: async (year, month) => {
    const response = await api.get('/reports/monthly', { params: { year, month } })
    return unwrap(response)
  },
}
